### PR TITLE
Treat plugins in collections whose name starts with `_` as private

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
~~Right now all plugins whose name starts with `_` are treated as deprecated by `ansible-doc -l` and by the validate-modules sanity test. However they are not shown as deprecated in ansible-doc itself.~~ (This has been changed in #79362.)

This PR uses the `_` prefix **outside of ansible.builtin and ansible.legacy** to mean private. Thus such plugins will not be shown until the user uses extra verbosity (`-v`).

##### ISSUE TYPE
- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME
ansible-doc

### CODE REVIEW

**Comment by mattclay** [comment]
@felixfontein Can you add this to the agenda for the core meeting?

**Comment by felixfontein** [comment]
@mattclay I added it. Unfortunately I won't be around, since I have something else at the time of the core meeting (for basically every core meeting).

File: lib/ansible/cli/doc.py
Comment (s-hertel) [inline lib/ansible/cli/doc.py:793]: I would rather not overload verbosity for something that is shown with `display.display()` rather than `display.v()`.

Besides my feeling that this is overloading `-v`, listing private content doesn't seem like something we should support. Instead, the internal names should be looked up explicitly if the documentation is needed.

**Review by s-hertel — CHANGES_REQUESTED** [review]
It seems easiest just to have an asynchronous conversation here. I think there is general consensus by the core team that we would like to avoid encoding meaning into filenames and use metadata to mark plugins as private for ansible-doc instead. This could be a new key in the runtime.yml, which would keep some possibilities open (if at some point we wanted to change runtime behavior for private plugins rather than just hide from ansible-doc).

File: lib/ansible/cli/doc.py
Comment (felixfontein) [reply lib/ansible/cli/doc.py:793]: I got inspired to only show these when `-v` is specified by https://github.com/ansible/ansible/pull/75116#discussion_r997335620. But I'm happy with not doing that :)

Comment (s-hertel) [reply lib/ansible/cli/doc.py:793]: Maybe that could be a bit surprising too :thinking:

**Comment by felixfontein** [comment]
I created #79362 which contains the changes that remove `_` handling in ansible-doc and validate-modules for collection plugins. I think this is something that should be removed no matter whether underscores are repurposed or not.

I'll create another new PR for extending meta/runtime.yml metadata to include a `private` flag, though probably only tomorrow or so.

**Comment by felixfontein** [comment]
The new PR is #79370.

**Comment by felixfontein** [comment]
Rebased after #79362 has been merged.

**Pull request closed without merging**
